src/PIL/PngImagePlugin.py

```diff
@@ -981,7 +981,16 @@ def load_end(self):
             except EOFError:
                 if cid == b"fdAT":
                     length -= 4
-                ImageFile._safe_read(self.fp, length)
+                try:
+                    ImageFile._safe_read(self.fp, length)
+                except OSError as e:
+                    if (
+                        str(e) == "Truncated File Read"
+                    ) and ImageFile.LOAD_TRUNCATED_IMAGES:
```

Comment:
```suggestion
                    if ImageFile.LOAD_TRUNCATED_IMAGES:
```
Given that [`ImageFile._safe_read()`](https://github.com/python-pillow/Pillow/blob/067c5f4123c7cab7507fca02605bfd9762861d1f/src/PIL/ImageFile.py#L572-L604) only explicitly raises an `OSError` with this message, I don't think the message needs to be checked.